drivers/led/lp5562.c

```diff
@@ -921,18 +926,101 @@ static int lp5562_led_update_current(const struct device *dev)
 	return ret;
 }
 
+static int lp5562_enable(const struct device *dev, bool soft_reset)
+{
+	const struct lp5562_config *config = dev->config;
+	const struct gpio_dt_spec *en_pin = &config->en_pin;
+	int err = 0;
+
+	/* if gpio control is enabled, we need to assert EN_GPIO first */
+	if (en_pin->port != NULL) {
+		err = gpio_pin_set_dt(en_pin, 1);
+		if (err) {
+			LOG_ERR("LP5562 LED chip EN_GPIO Assert Failed.");
+			return err;
+		}
+		/*
+		 * The I2C host should allow at least 1ms before sending data to the LP5562
+		 * after the rising edge of the enable line
+		 */
+		k_sleep(K_MSEC(1));
+	}
+
+	if (soft_reset) {
+		/* reset all internal registers to have a deterministic state */
+		err = i2c_reg_write_byte_dt(&config->bus, LP5562_RESET, 0xFF);
+		if (err) {
+			LOG_ERR("LP5562 LED chip Soft-Reset failed.");
+			return err;
+		}
+	}
+
+	/* set en bit in register configurations */
+	err = i2c_reg_update_byte_dt(&config->bus, LP5562_ENABLE, LP5562_ENABLE_CHIP_EN_MASK,
+				     LP5562_ENABLE_CHIP_EN_SET);
+	if (err) {
+		LOG_ERR("LP5562 LED chip Enable failed.");
+		return err;
+	}
+	/* Allow 500 µs delay after setting chip_en bit to '1' */
+	k_sleep(K_MSEC(1));
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int lp5562_disable(const struct device *dev)
+{
+	const struct lp5562_config *config = dev->config;
+	const struct gpio_dt_spec *en_pin = &config->en_pin;
```

Comment:
Shouldn't there be a NULL check? or do we assume that's handled by writing to the device?

Reply:
I didn't get it, which NULL check is missing could you explain?

```
	const struct lp5562_config *config = dev->config;
	const struct gpio_dt_spec *en_pin = &config->en_pin;
```
these lines are used everywhere without check and `en_pin` handling is inside NULL check.